### ISSUE TITLE
JS performance: flarum/utils/ScrollListener.loop()

### ISSUE THREAD

**Issue by BartVB** [flarum/core#1222]
For some reason a tab in Chrome 59 with a Flarum.org discussion that I don't interact with uses approx 4% CPU on my MBP Pro 2014. Not a lot but this shouldn't be necessary and this is probably quite a drain on mobile devices. With the Chrome Inspector Performance pane it was quite trivial to pinpoint this to:

[flarum/utils/ScrollListener.loop()](https://datitisev.github.io/FlarumAPIDocs/js/master/class/js/lib/utils/ScrollListener.js~ScrollListener.html)

Apparently that method is called on each animation frame and for some reason this is triggered even for completely idle pages. Alas I'm not exactly a JS expert so I don't have a clue as to why this is happening or what can be done about this.

**Comment by franzliedke** [comment]
@tobscure I faintly remember that we once talked about this implementation, and why it was necessary (?). Why, for example, do we not just listen for "scroll" events on the window?

**Comment by tobyzerner** [comment]
https://developer.mozilla.org/en-US/docs/Web/API/Document/scroll_event

**Comment by JoshyPHP** [comment]
Coincidentally, I was looking at `MutationObserver` and how often it would trigger on an application such as Flarum and I noticed something strange. Every scrolling event removes then re-adds identical nodes in the scrubber. For instance, you're at "40 of 46 posts" in a thread, you scroll down 1 pixel and you're still at "40 of 46 posts" but the text node for "40" is not the same anymore. I suppose there's some DOM diffing going on that ends up replacing text nodes even though their content hasn't changed, or something like that. I didn't try to debug it because that's not something I'm familiar with at all.

For reference, here's the snippet I entered in my browser's console to monitor changes:
```js
new MutationObserver(console.dir).observe(document.body, {"childList":true, "subtree":true});
```

Quick test that shows that the text node for the scrubber is not the same:
```js
t1 = document.querySelector('span.Scrubber-index').firstChild;
window.scrollBy(0, 1);
setTimeout(function(){
	t2 = document.querySelector('span.Scrubber-index').firstChild;
	console.log(t1.isEqualNode(t2) + ' ' + t1.isSameNode(t2));
}, 10);
```
